### PR TITLE
feat(debug-screen): add debug screen with mls data (AR-1778)

### PR DESCRIPTION
**Pull request by mchenani**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1778" title="AR-1778" target="_blank">SubtaskAR-1778</a>  Show KeyPackages count and status in Log Screen
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Created a new debug-screen and added MLS keyPackagesCount, MLSClientId, and also moved the Log settings from the settings screen to it.
[AR-1778]

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

We need to see how many keypackages do you have fro MLS till we add the usecase to manage them and topup new keypackages. 

### Solutions

Created a debug screen for future needed to add other stuff into it for debugging purposes.

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [x] GitHub link to other pull request
[LinkedPR-KeyPackageCountsUseCase]

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Manual Test: from the app menu, you can access the new debug screen.

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[LinkedPR-KeyPackageCountsUseCase]: https://github.com/wireapp/kalium/pull/591
[AR-1778]:https://wearezeta.atlassian.net/browse/AR-1778

[AR-1778]: https://wearezeta.atlassian.net/browse/AR-1778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    val debugScreenViewModel: DebugScreenViewModel = hiltViewModel()
+    Column {
+        topBar("Debug")
+        list("MLS Data") { debugScreenViewModel.mlsData.map { textRowItem(it) } }
+        list("Logs") { debugLog(debugScreenViewModel) }
+    }
+}
+
+@Composable
+fun topBar(title: String) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:45]: I'm not sure, but I think we discussed the topic of compose function names, and I feel like we agreed to treat them like these sort of constructors and we usually give them uppercase names 🤔

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    headerTitle: String, content: @Composable () -> Unit = {}
+) {
+    Column {
+        FolderHeader(headerTitle)
+        content()
+    }
+}
+
+@Composable
+fun textRowItem(text: String, trailingIcon: Int? = null, onIconClick: () -> Unit = {}) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:69]: Can `trailingIcon` be annotated with `@DrawableRes`?

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    Row(modifier = Modifier.fillMaxWidth().background(Color.White)) {
+        Text(
+            text = text,
+            fontWeight = FontWeight.Normal,
+            color = Color.DarkGray,
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:74]: Do we care about light/dark theme compatibility here? 😄 if so, then colors should be theme-dependent

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            modifier = Modifier.padding(12.dp).weight(1f),
+            textAlign = TextAlign.Left,
+            fontSize = 14.sp
+        )
+        trailingIcon?.let {
+            Icon(painter = painterResource(id = trailingIcon),
+                contentDescription = "",
+                tint = Color.Black,
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:82]: Same here, otherwise it won't be visible on dark theme

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-669/3/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+) {
+    Column {
+        FolderHeader(headerTitle)
+        content()
+    }
+}
+
+@Composable
+fun TextRowItem(text: String, @DrawableRes trailingIcon: Int? = null, onIconClick: () -> Unit = {}) {
+    Row(modifier = Modifier.fillMaxWidth().background(Color.White)) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:71]: Can you also change the background color to be for instance `MaterialTheme.colorScheme.surface`? 😃

File: app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        "Share Logs",
+        trailingIcon = android.R.drawable.ic_menu_share
+    ) { context.startFileShareIntent(debugScreenViewModel.logFilePath(absolutePath)) }
+}
+
+@Composable
+fun SwitchRowItem(
+    text: String, checked: Boolean = false, onCheckedChange: ((Boolean) -> Unit)?
+) {
+    Row(modifier = Modifier.fillMaxWidth().background(Color.White)) {
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/debugscreen/DebugScreen.kt:110]: Can you also change the background color to be for instance `MaterialTheme.colorScheme.surface`? 😃

File: app/src/main/kotlin/com/wire/android/ui/settings/SettingsScreen.kt [hunk 7]
```diff
@@ -2,105 +2,34 @@ package com.wire.android.ui.settings
...
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(MaterialTheme.wireDimensions.spacing16x)
-        )
-
-
         Text(
             text = "Settings is under construction",
             fontWeight = FontWeight.Bold,
             color = Color.Black,
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/settings/SettingsScreen.kt:27]: Can you also change this color to be for instance `MaterialTheme.colorScheme.onBackground`? 😃

**Review by saleniuk — APPROVED** [review]
Just 3 other places where we currently have hardcoded color, but generally nice job! 👍

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-669/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-669/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-669/8/) **failed**.

**Comment by AndroidBob** [comment]
Build [11](https://10.10.124.134/job/wire-android-reloaded/job/PR-669/11/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3640733)(3640733)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3640733)(3640733)-dev-release.apk)

**Pull request merged**